packages/devtools_app/lib/src/debugger/flutter/codeview.dart

```diff
@@ -185,52 +197,76 @@ class GutterRow extends StatelessWidget {
     Key key,
     @required this.lineNumber,
     @required this.totalLines,
-    @required this.onPressed,
     @required this.isBreakpoint,
-    this.isPausedHere = false,
+    @required this.isExecutable,
+    @required this.isPausedHere,
+    @required this.onPressed,
   }) : super(key: key);
 
   final int lineNumber;
   final int totalLines;
-  final VoidCallback onPressed;
-
   final bool isBreakpoint;
+  final bool isExecutable;
 
   /// Whether the execution point is currently paused here.
   final bool isPausedHere;
 
+  final VoidCallback onPressed;
+
   @override
   Widget build(BuildContext context) {
-    // TODO(devoncarew): Improve the layout of the breakpoint and execution
-    // marker.
+    final theme = Theme.of(context);
+
+    final foregroundColor = theme.textTheme.bodyText2.color;
+    final subtleColor = theme.unselectedWidgetColor;
 
     return InkWell(
       onTap: onPressed,
       child: Container(
         height: CodeView.rowHeight,
-        padding: const EdgeInsets.only(left: 2.0, right: 4.0),
-        alignment: Alignment.centerRight,
+        padding: const EdgeInsets.only(right: 4.0),
         decoration: BoxDecoration(color: titleSolidBackgroundColor),
-        child: Row(
+        child: Stack(
+          alignment: AlignmentDirectional.centerStart,
+          fit: StackFit.expand,
           children: [
+            if (isExecutable || isBreakpoint)
+              Align(
+                alignment: Alignment.centerLeft,
+                child: SizedBox(
+                  width: 12,
+                  height: 12,
+                  child: Stack(
+                    alignment: AlignmentDirectional.center,
+                    children: [
+                      if (isExecutable) _createCircle(1.5, subtleColor),
+                      if (isBreakpoint) _createCircle(6.0, foregroundColor),
+                    ],
+                  ),
+                ),
+              ),
+            Text('$lineNumber', textAlign: TextAlign.end),
             if (isPausedHere)
-              const Icon(Icons.play_arrow, size: defaultIconSize),
-            if (isBreakpoint && !isPausedHere)
               const Padding(
-                padding: EdgeInsets.only(bottom: 4.0),
-                child: Text('●'),
-              ),
-            Expanded(
-              child: Text(
-                '$lineNumber',
-                textAlign: TextAlign.end,
+                padding: EdgeInsets.only(left: 8.0),
```

Comment:
use const from theme file

Reply:
This is for padding to avoid the breakpoint area on the left - I'll add a specific constant for that (we'll want this to change independently of the values in the theme file).